### PR TITLE
Mock requests in storybook

### PR DESCRIPTION
**Pull request by woxxy**

### CODE REVIEW

File: src/components/molecules/FollowerGoal.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { useTwitchFollowers } from '../../hooks/twitch';
 import { LoadingBarClassNames, LoadingBar } from './internal/LoadingBar';
 
-interface FollowerGoalProps {
+export interface FollowerGoalProps {
```
Comment (woxxy) [inline src/components/molecules/FollowerGoal.tsx]: Stories would cry if this is not exported

File: src/interfaces.tsx [hunk 2]
```diff
@@ -47,6 +47,8 @@ interface TwitchBroadcasterSubscriber {
   broadcaster_id: string;
   broadcaster_name: string;
   is_gift: boolean;
+  gifter_id: string;
```
Comment (woxxy) [inline src/interfaces.tsx]: These aren't mentioned in the Twitch API but are part of the payload

File: .storybook/main.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  stories: ['../src/**/*.stories.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
```
Comment (nstepien) [inline .storybook/main.js]: Let's remove those mdx files.
```suggestion
  stories: ['../src/**/*.stories.*'],
```

File: .storybook/main.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+  addons: ['@storybook/add on-links', '@storybook/addon-essentials'],
```
Comment (nstepien) [inline .storybook/main.js]: Typo?
```suggestion
  addons: ['@storybook/addon-links', '@storybook/addon-essentials'],
```

File: .storybook/main.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+};
```
Comment (nstepien) [inline .storybook/main.js:8]:
```suggestion
  reactOptions: {
    fastRefresh: true,
    strictMode: true
  }
};
```

File: .storybook/preview.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { TwitchContext } from '../src/components/TwitchContext';
+import React from 'react';
```
Comment (nstepien) [inline .storybook/preview.js]:
```suggestion
```

File: .storybook/preview.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { TwitchContext } from '../src/components/TwitchContext';
```
Comment (nstepien) [inline .storybook/preview.js]: Please rename this file to `preview.tsx`, it should just work.

File: .storybook/preview.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import React from 'react';
+
+export const parameters = {
+  actions: { argTypesRegex: '^on[A-Z].*' },
+};
```
Comment (nstepien) [inline .storybook/preview.js]: What does this do?

File: src/components/molecules/FollowerGoal.tsx (hunk 1 above)
Comment (nstepien) [reply src/components/molecules/FollowerGoal.tsx]: It works fine here without exporting it, can you review those added exports?

File: src/stories/MoleculeFollowerGoal.tsx [hunk 9]
```diff
@@ -1,14 +1,8 @@
 import { ComponentProps } from 'react';
-import './button.css';
-import { TwitchProvider } from '../components/TwitchProvider';
-import { FollowerGoal } from '../components/molecules/FollowerGoal';
+import { FollowerGoal } from '../components/molecules';
 
 export const MoleculeFollowerGoalStory = (
```
Comment (nstepien) [inline src/stories/MoleculeFollowerGoal.tsx]: Can we just do this?
```suggestion
export const MoleculeFollowerGoalStory = FollowerGoal;
```

**Review by nstepien — CHANGES_REQUESTED** [review]
Please remove the other irrelevant template files under `src/stories/`.
Can you also add a CI step to build the storybook, make sure we don't break anything between commits.

File: .storybook/main.js (hunk 4 above)
Comment (woxxy) [reply .storybook/main.js]: Probably typed space by mistake yeah

File: .storybook/preview.js (hunk 8 above)
Comment (woxxy) [reply .storybook/preview.js]: Actually part of the default, I haven't looked into it yet

File: src/stories/MoleculeFollowerGoal.tsx (hunk 9 above)
Comment (woxxy) [reply src/stories/MoleculeFollowerGoal.tsx]: We can probably just avoid this entire file if we're just returning the same thing

Comment (woxxy) [reply src/stories/MoleculeFollowerGoal.tsx]: https://storybook.js.org/docs/react/essentials/actions Seems to be related to making a log for actions like onClick when used in the interface. Looks sensible to leave this alone if we make something reliant on actions.

File: .storybook/preview.js (hunk 8 above)
Comment (woxxy) [reply .storybook/preview.js]: https://storybook.js.org/docs/react/essentials/actions Seems to be related to making a log for actions like onClick when used in the interface. Looks sensible to leave this alone if we make something reliant on actions.

File: src/components/molecules/FollowerGoal.tsx (hunk 1 above)
Comment (woxxy) [reply src/components/molecules/FollowerGoal.tsx]: image
I get this issue from IntelliJ if I don't export it. I don't get this error with the `typecheck` command and everything works, so I am puzzled now.

File: src/stories/mock/index.tsx [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import fetchMock from 'fetch-mock';
+export { followers } from './followers';
+export { subscriber } from './subscriber';
+export { users } from './users';
+export default fetchMock;
```
Comment (nstepien) [inline src/stories/mock/index.tsx]: Quick simplification
```suggestion
export { default } from 'fetch-mock';
export { followers } from './followers';
export { subscriber } from './subscriber';
export { users } from './users';
```

File: src/stories/MoleculeFollowerGoal.stories.tsx [hunk 11]
```diff
@@ -1,20 +1,23 @@
...
+  argTypes: {},
 } as Meta;
 
-const Template: Story<
-  ComponentProps<typeof MoleculeFollowerGoalStory>
-> = args => <MoleculeFollowerGoalStory {...args} />;
+const Template: Story<ComponentProps<typeof FollowerGoal>> = args => {
+  mock.restore();
+  mock.mock(`${TWITCH_API_ENDPOINT}users`, users.me);
+  mock.mock(`begin:${TWITCH_API_ENDPOINT}users/follows`, followers.me);
```
Comment (nstepien) [inline src/stories/MoleculeFollowerGoal.stories.tsx]: I think we can just move all the mocking setup in `preview.tsx`, so it doesn't have to be redone in all individual stories.

**Review by nstepien — CHANGES_REQUESTED** [review]

File: src/stories/MoleculeFollowerGoal.stories.tsx (hunk 11 above)
Comment (woxxy) [reply src/stories/MoleculeFollowerGoal.stories.tsx]: Done

**Review by nstepien — APPROVED** [review]

**Pull request merged**